ext/AvizExt/viz/clustering.jl

```diff
@@ -14,56 +16,91 @@ Visualize clustered time series of scenarios.
 Figure
 """
 function ADRIA.viz.clustered_scenarios(
-    data::AbstractMatrix,
+    data::NamedDimsArray,
     clusters::Vector{Int64};
     fig_opts::Dict=Dict(),
-    axis_opts::Dict=Dict()
-)
+    axis_opts::Dict=Dict(),
+    opts::Dict=Dict()
+)::Figure
     f = Figure(; fig_opts...)
     g = f[1, 1] = GridLayout()
 
-    ADRIA.viz.clustered_scenarios!(g, data, clusters; axis_opts=axis_opts)
+    ADRIA.viz.clustered_scenarios!(g, data, clusters; axis_opts=axis_opts, opts=opts)
 
     return f
 end
 function ADRIA.viz.clustered_scenarios!(
     g::Union{GridLayout,GridPosition},
-    data::AbstractMatrix,
+    data::NamedDimsArray,
     clusters::Vector{Int64};
-    axis_opts::Dict=Dict()
-)
-    # Ensure last year is always shown in x-axis
+    axis_opts::Dict=Dict(),
+    opts::Dict=Dict()
+)::Union{GridLayout,GridPosition}
     xtick_vals = get(axis_opts, :xticks, _time_labels(timesteps(data)))
     xtick_rot = get(axis_opts, :xticklabelrotation, 2 / π)
-    ax = Axis(
-        g[1, 1],
-        xticks=xtick_vals,
-        xticklabelrotation=xtick_rot;
-        axis_opts...
-    )
-
-    # Compute cluster colors
-    clusters_colors = _clusters_colors(clusters)
-    unique_cluster_colors = unique(clusters_colors)
-
-    leg_entry = Any[]
-    for cluster in unique(clusters)
-        cluster_color = _cluster_color(unique_cluster_colors, clusters, cluster)
-        push!(
-            leg_entry,
-            series!(
-                ax,
-                data[:, clusters.==cluster]',
-                solid_color=cluster_color
-            )
-        )
+    ax = Axis(g[1, 1], xticks=xtick_vals, xticklabelrotation=xtick_rot; axis_opts...)
+
+    if get(opts, :summarize, true)
+        _plot_clusters_confint!(g, ax, data, clusters; opts=opts)
+    else
+        _plot_clusters_series!(g, ax, data, clusters; opts=opts)
     end
 
-    # Plot Legend
+    return g
+end
+
+function _plot_clusters_series!(
+    g::Union{GridLayout,GridPosition},
+    ax::Axis,
+    data::NamedDimsArray,
+    clusters::Vector{Int64};
+    opts::Dict=Dict()
+)::Nothing
+    alphas = _get_alphas(clusters)
+    clusters_colors = unique(_get_colors(clusters, alphas))
+
+    leg_entry::Vector{Any} = [
+        series!(ax, data[:, clusters.==cluster]', solid_color=clusters_colors[idx_c])
+        for (idx_c, cluster) in enumerate(unique(clusters))
+    ]
+
     n_clusters = length(unique(clusters))
     Legend(g[1, 2], leg_entry, "Cluster " .* string.(1:n_clusters), framevisible=false)
 
-    return g
+    return nothing
+end
+
+function _plot_clusters_confint!(
+    g::Union{GridLayout,GridPosition},
+    ax::Axis,
+    data::NamedDimsArray,
+    clusters::Vector{Int64};
+    opts::Dict=Dict()
+)::Nothing
+    band_colors = unique(_get_colors(clusters, 0.5))
+    line_colors = unique(_get_colors(clusters))
+    legend_entry = Vector{Any}(undef, length(unique(clusters)))
+
+    x_timesteps::UnitRange{Int64} = 1:length(data.timesteps)

```

Comment:
```suggestion
    x_timesteps::UnitRange{Int64} = 1:length(timesteps(data))

```

Reason we'd want to rely on an interface is in case we change the underlying object/type or dimension name.

This might happen as we make the visualizations "smarter" to get plots to show additional info, like we spoke about (display name of metric, unit of measure, etc).